compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -516,7 +516,8 @@ class Typer extends Namer
           case TypeApplications.EtaExpansion(tycon) => tpt1 = tpt1.withType(tycon)
           case _ =>
         }
-        checkClassType(tpt1.tpe, tpt1.pos, traitReq = false, stablePrefixReq = true)
+        if (checkClassType(tpt1.tpe, tpt1.pos, traitReq = false, stablePrefixReq = true) eq defn.ObjectType)
+          tpt1 = TypeTree(defn.ObjectType)
```

Comment:
Hm, creating a new tree like this will not set its position and so on, unlike `tpt1 = tpt1.withType(defn.ObjectType)` (like here) or `TypeTree(defn.ObjectType) withPos tpt.pos` (like in `New`). OTOH, this isn't a tree the user wrote. And maybe the point of using `defn.ObjectType` is to hope we get no other error here?

What makes more sense @allanrenucci?

Reply:
`tpt1 = TypeTree(defn.ObjectType).withPos(tpt1.pos)` I think